_posts/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment.md

```diff
@@ -0,0 +1,396 @@
+| Machine | Cluster API resource for managing nodes - Controlplane or Workers. In case of Controlplane, its directly managed by `KubeadmControlPlane`, whereas for Workers it's managed by a `MachineSet`|
+| Metal3Machine | Corresponding Metal3 resource for managing Baremetal Nodes, it's managed by a `Machine` resource |
+| Metal3MachineTemplate | Metal3 resource which acts as a template when creating a Controlplane or a Worker node |
+| KubeadmConfigTemplate | A template of `KubeadmConfig`, for Workers, used to generate KubeadmConfig when a new worker node is provisioned |
+
+> **Note** : The corresponding `KubeadmConfig` is copied to the Controlplane/Worker at the time of provisioning.  
+  
+<br/>
+
+
+## BareMetal Cluster Deployment
+
+The deployment scripts primarily use ansible and the existing Kubernetes management cluster (based on minikube ) for deploying the bare-metal cluster. Make sure that some of the environment variables used for Metal³ deployment are set, if you didn't use `config_${user}.sh` for setting the environment variables.
+
+| Parameter           | Description                  | Default                  |
+| ------------------- | ---------------------------- | ------------------------ |
+| CAPI_VERSION        | Version of Metal3 API4       | v1alpha3        |
+| POD_CIDR            | Pod Network CIDR             | 192.168.0.0/18           |
+| CLUSTER_NAME        | Name of bare metal cluster   | test1                    |
+
+<br/>
+
+
+### Steps Involved
+
+All the scripts for cluster provisioning or deprovisioning are located at - [`${metal3-dev-env}/scripts/v1alphaX/`](https://github.com/metal3-io/metal3-dev-env/tree/master/scripts/v1alphaX). The scripts call a common playbook which handles all the tasks that are available.
+
+> info "Note"
+> Here is a depiction of the common steps, mainly involving generating templates and config files. 
+
+<br/>
+
+!["A diagram depicting the Generate Templates workflow"](/assets/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment/metal3-generate-templates.png)
+<br/>
+<br/>
+
+The steps involved in the process are :
+
+- The script calls an ansible playbook with necessary parameter ( from env variables and defaults )
+- The playbook executes the role -, [`${metal3-dev-env}/vm-setup/roles/v1aX_integration_test`](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test), which runs the corresponding [task_file](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/tasks) for provisioning/deprovisioning the cluster/controlplane or a worker
+- There are [templates](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/templates) in the role, which are copied to the `manifest` directory. They are used for generating configuration for the cluster and kubeadm, which is then supplied to the Kubernetes module of ansible to create the cluster.
```

Comment:
I would rather use the term "rendered" instead of "copied". for example, the clusterctl templates are different depending on the version you deploy with (v1a3 or v1a4).

Reply:
I have updated this part according the PR you submitted last week.